recipes/stx/all/conanfile.py

```diff
@@ -0,0 +1,142 @@
+    settings = 'os', 'compiler', 'build_type', 'arch'
+    options = {
+        'backtrace': [True, False],
+        'panic_handler': [None, 'default', 'backtrace'],
+        'shared': [True, False],
+        'fPIC': [True, False],
+        'visible_panic_hook': [True, False],
+    }
+    default_options = {
+        'backtrace': False,
+        'panic_handler': 'default',
+        'shared': False,
+        'fPIC': True,
+        'visible_panic_hook': False,
+    }
+    exports_sources = ['CMakeLists.txt', 'patches/*']
+
+    def config_options(self):
+        if self.settings.os == 'Windows':
+            del self.options.fPIC
+
+    def configure(self):
+        if (self.options.panic_handler == 'backtrace' and
+                not self.options.backtrace):
+            raise ConanInvalidConfiguration(
+                'panic_handler=backtrace requires backtrace=True'
+            )
+
+        compiler = self.settings.compiler
+        compiler_version = tools.Version(self.settings.compiler.version)
+
+        if compiler.get_safe('cppstd'):
+            tools.check_min_cppstd(self, 17)
+
+        if compiler == 'Visual Studio' and compiler_version < 16:
+            raise ConanInvalidConfiguration(
+                'STX requires C++17 support, use at least VS 2019'
+            )
+
+        if compiler == 'gcc' and compiler_version < 8:
+            raise ConanInvalidConfiguration(
+                'STX requires C++17 support, use at least GCC 8'
+            )
+
+        if (compiler == 'clang' and compiler.libcxx and
+                compiler.libcxx in ['libstdc++', 'libstdc++11'] and
+                compiler_version < 9):
+            raise ConanInvalidConfiguration(
+                'STX requires C++17 language and standard library features '
+                'which clang & libc++ < 10 lack'
+            )
+
+        if (compiler == 'clang' and compiler.libcxx and
+                compiler.libcxx == 'libc++' and
+                compiler_version < 10):
+            raise ConanInvalidConfiguration(
+                'STX requires C++17 language and standard library features '
+                'which clang & libc++ < 10 lack'
+            )
+
+        if compiler == 'apple-clang':
+            raise ConanInvalidConfiguration(
+                'STX requires C++17 language and standard library features '
+                'which apple-clang and libc++ lack'
+            )
+
+        if (compiler == 'Visual Studio' and self.options.shared and
+                tools.Version(self.version) <= '1.0.1'):
+            raise ConanInvalidConfiguration(
+                'shared library build does not work on windows with '
+                'STX version <= 1.0.1'
+            )
+
+    def requirements(self):
+        if self.options.backtrace:
+            self.requires('abseil/20200923.1')
+
+    def source(self):
+        tools.get(**self.conan_data['sources'][self.version])
+        tools.rename(src=f'STX-{self.version}', dst='source_subfolder')
```

Comment:
```suggestion
        tools.rename('STX-{}'.format(self.version), dst='source_subfolder')
```

String format is not allowed, all recipes should be compatible with old python versions